### PR TITLE
feat(unified-share-modal): add contextual callout tooltips

### PR DESCRIPTION
**Pull request by ittaishay**
### `shared-link-toggle`
Screen Shot 2019-03-19 at 4 07 40 PM

### `shared-link-access-menu`
Screen Shot 2019-03-19 at 4 06 36 PM

### `shared-link-settings`
Screen Shot 2019-03-19 at 4 07 25 PM

### `shared-link-copy-button`
Screen Shot 2019-03-19 at 4 07 00 PM

### Multiple at once
Screen Shot 2019-03-19 at 4 08 11 PM

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @ittaishay has signed the CLA. Thanks for the pull request!

File: src/components/tooltip/Tooltip.js [hunk 1]
```diff
@@ -79,6 +79,8 @@ type Props = {
     constrainToWindow: boolean,
     /** Forces the tooltip to be shown or hidden (useful for errors) */
     isShown?: boolean,
+    /** Function called if the user manually dismisses the tooltip - only applies if showCloseButton is true */
+    onDismiss?: () => void,
```
Comment (drichard-box) [inline src/components/tooltip/Tooltip.js]: onClose then?

File: src/features/unified-share-modal/SharedLinkAccessMenu.js [hunk 2]
```diff
@@ -23,14 +23,16 @@ type Props = {
     classificationName?: string,
     enterpriseName?: string,
     itemType: ItemType,
+    onDismissTooltip: (componentIdentifier: tooltipComponentIdentifierType) => void,
```
Comment (drichard-box) [inline src/features/unified-share-modal/SharedLinkAccessMenu.js]: remove the param?

File: src/features/unified-share-modal/UnifiedShareModal.js [hunk 3]
```diff
@@ -443,11 +449,12 @@ class UnifiedShareModal extends React.Component<Props, State> {
             </div>
         );
 
+        const hasOtherTooltip = Object.keys(tooltips).length > 0;
```
Comment (drichard-box) [inline src/features/unified-share-modal/UnifiedShareModal.js]: lets leave this up to the caller and show all tooltips at once

File: src/features/unified-share-modal/flowTypes.js [hunk 4]
```diff
@@ -140,3 +140,9 @@ export type collaboratorType = {
 export type collaboratorsListType = {
     collaborators: Array<collaboratorType>,
 };
+
+export type tooltipComponentIdentifierType =
```
Comment (drichard-box) [inline src/features/unified-share-modal/flowTypes.js]: just componentIdentifierType?

Comment (drichard-box) [reply src/features/unified-share-modal/flowTypes.js]: nvm this makes more sense since it restricts the possible values for the tooltip prop

**Pull request merged**